fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/NumberParmBehaviorUtil.java

```diff
@@ -0,0 +1,289 @@
+
+                        bindVariables.add(generateLowerBound(originalNumber));
+                        bindVariables.add(generateUpperBound(originalNumber));
+
+                        // <CODE>BASIC_NUMBER_VALUE <= ? OR BASIC_NUMBER_VALUE > ?</CODE> 
+                        whereClauseSegment.append(tableAlias + DOT).append(NUMBER_VALUE);
+                        whereClauseSegment.append(JDBCOperator.LTE.value())
+                                .append(BIND_VAR);
+                        whereClauseSegment.append(JDBCConstants.SPACE).append(JDBCOperator.OR)
+                                .append(JDBCConstants.SPACE);
+                        whereClauseSegment.append(tableAlias + DOT).append(NUMBER_VALUE);
+                        whereClauseSegment.append(JDBCOperator.GT.value())
+                                .append(BIND_VAR);
+                    }
+                    break;
+                case AP:
+                    // ap - approximate
+                    // -10%
+                    bindVariables.add(originalNumber.multiply(LOWER_BOUND));
+                    // +10%
+                    bindVariables.add(originalNumber.multiply(UPPER_BOUND));
+
+                    // <CODE>BASIC_NUMBER_VALUE >= ? AND BASIC_NUMBER_VALUE <= ?</CODE> 
+                    whereClauseSegment.append(tableAlias + DOT).append(NUMBER_VALUE);
+                    whereClauseSegment.append(JDBCOperator.GTE.value())
+                            .append(BIND_VAR);
+                    whereClauseSegment.append(JDBCConstants.SPACE).append(JDBCOperator.AND).append(JDBCConstants.SPACE);
+                    whereClauseSegment.append(tableAlias + DOT).append(NUMBER_VALUE);
+                    whereClauseSegment.append(JDBCOperator.LTE.value())
+                            .append(BIND_VAR);
+                    break;
+                default:
+                    // gt, lt, ge, le, sa, eb
+                    // take the default behavior.
+                    // Build this piece: p1.value_string {operator} search-attribute-value
+                    JDBCOperator operator = queryBuilder.getPrefixOperator(value);
+                    whereClauseSegment.append(tableAlias + DOT).append(NUMBER_VALUE).append(operator.value())
+                            .append(BIND_VAR);
+                    bindVariables.add(originalNumber);
+                    break;
+                }
+            }
+        }
+
+        // End the Clause started above, and closes the parameter expression. 
+        // Query: )) 
+        whereClauseSegment.append(RIGHT_PAREN).append(RIGHT_PAREN);
+    }
+
+    public static BigDecimal generateLowerBound(BigDecimal original) {
+
+        int factor = calculateSignificantFigures(original);
+        if (factor < 0) {
+            factor = Math.abs(factor) - 1;
+        } else {
+            factor = original.scale() + 1;
+
+            // Mutate it to the ORIGINAL string
+            String t = "" + original;
```

Comment:
will this always print it in scientific notation?

Reply:
No, it actually prints as the original